File: C/FizzBuzz/fizzBuzzTest.c
#include <string.h>
#include<assert.h>

void fizzBuzz(int num, char* output);

int main(void) {
    char output[10];
    fizzBuzz(0, output);
    assert(strcmp(output, "0") == 0);
    fizzBuzz(1, output);
    assert(strcmp(output, "1") == 0);
    fizzBuzz(3, output);
    assert(strcmp(output, "Fizz") == 0);
    fizzBuzz(5, output);
    assert(strcmp(output, "Buzz") == 0);
    fizzBuzz(15, output);
    assert(strcmp(output, "FizzBuzz") == 0);
    return 0;
}

File: Java/CyberDojo/fizzBuzz/src/FizzBuzzTest.java
import org.junit.jupiter.api.*;
import static org.junit.jupiter.api.Assertions.*;
import org.junit.Test;

public class FizzBuzzTest {
    @Test
    public void testFizzBuzz() {
        assertEquals("0", FizzBuzz.fizzBuzz(0));
        assertEquals("1", FizzBuzz.fizzBuzz(1));
        assertEquals("Fizz", FizzBuzz.fizzBuzz(3));
        assertEquals("Buzz", FizzBuzz.fizzBuzz(5));
        assertEquals("FizzBuzz", FizzBuzz.fizzBuzz(15));
    }

    @Test(expected = IllegalArgumentException.class)
    public void testNegativeNumbers() {
        FizzBuzz.fizzBuzz(-1);
    }

}
